lib/dgram.js

```diff
@@ -346,6 +346,9 @@ Socket.prototype.send = function(buffer,
   if (self._bindState == BIND_STATE_UNBOUND)
     self.bind({port: 0, exclusive: true}, null);
 
+  if (buffer.length === 0)
+    buffer[0] = Buffer.allocUnsafe(0);
```

Comment:
It's more of a personal preference but can you use `buffer.push(...)`?  That makes it clear we're appending, not replacing, without having to look at the surrounding code.